doc/whats_new/v1.4.rst

```diff
@@ -93,10 +93,11 @@ Changelog
 ...................
 
 - |Fix| :class:`tree.DecisionTreeClassifier` and
-  :class:`tree.DecisionTreeRegressor` are handling missing values properly. The internal
-  criterion was not initialize when no missing values were present in the data, leading
-  to potentially wrong criterion values.
-  :pr:`28295` by :user:`Guillaume Lemaitre <glemaitre>`.
+  :class:`tree.DecisionTreeRegressor` are handling missing values properly for ``squared_error``
+  and ``friedman_mse`` criterion. The internal criterion was not initialize when no missing
+  values were present in the data, and did not always account for missing values for squared
+  error based criterion, leading to potentially wrong criterion values.
```

Comment:
You can let the previous description because all criterion were wrong. The one that are correct. The one that were correct did not support missing values :)

Reply:
Reverted